src/content/docs/fr/core-concepts/endpoints.mdx

```diff
@@ -0,0 +1,221 @@
+  return new Response(
+    JSON.stringify({
+      name: 'Astro',
+      url: 'https://astro.build/'
+    })
+  )
+}
+```
+
+Depuis Astro v3.0, l'objet `Response` retourné n'a plus besoin d'inclure la propriété `encoding`. Par exemple, pour produire une image png binaire :
+
+```ts title="src/pages/astro-logo.png.ts" {3}
+export async function GET({ params, request }) {
+  const response = await fetch("https://docs.astro.build/assets/full-logo-light.png");
+  return new Response(await response.arrayBuffer());
+}
+```
+
+Vous pouvez aussi typer vos fonctions d'endpoints en utilisant le type `APIRoute` :
+
+```ts
+import type { APIRoute } from 'astro';
+
+export const GET: APIRoute = async ({ params, request }) => {...}
+```
+
+### `params` et routage dynamique
+
+Les endpoints supportent les même fonctions de [routage dynamique](/fr/core-concepts/routing/#routes-dynamiques) que celles des pages. Nommez votre fichier avec un nom de paramètre entre crochets et exportez une fonction [`getStaticPaths()`](/fr/reference/api-reference/#getstaticpaths). Vous pouvez ensuite accéder au paramètre en utilisant la propriété `params` passée à la fonction de l'endpoint :
+
+```ts title="src/pages/api/[id].json.ts"
+import type { APIRoute } from 'astro';
+
+const usernames = ["Sarah", "Chris", "Yan", "Elian"]
+
+export const GET: APIRoute = ({ params, request }) => {
+  const id = params.id;
+  return new Response(
+    JSON.stringify({
+      name: usernames[id]
+    })
+  )
+}
+
+export function getStaticPaths() {
+  return [ 
+    { params: { id: "0"} },
+    { params: { id: "1"} },
+    { params: { id: "2"} },
+    { params: { id: "3"} }
+  ]
+}
+```
+
+Cela va générer 4 points de terminaison JSON au moment du build : `/api/0.json`, `/api/1.json`, `/api/2.json` et `/api/3.json`. Le routage dynamique avec les endpoints fonctionne de la même manière qu'avec les pages, mais parce que l'endpoint est une fonction et non un composant, les [props](/fr/reference/api-reference/#data-passing-with-props) ne sont pas supportées.
+
+### `request`
+
+Tous les endpoints reçoivent une propriété `request`, mais dans le mode statique, vous n'avez accès qu'à `request.url`. Cela renvoie l'URL complète du point de terminaison actuel et fonctionne de la même manière que [Astro.request.url](/fr/reference/api-reference/#astrorequest) pour les pages.
+
+```ts title="src/pages/request-path.json.ts"
+import type { APIRoute } from 'astro';
+
+export const GET: APIRoute = ({ params, request }) => {
+  return new Response(JSON.stringify({
+      path: new URL(request.url).pathname
+    })
+  )
+}
+```
+
+## Endpoints du serveur (Routes API)
+
+Tout ce qui est décrit dans la section sur les endpoints des fichiers statiques peut également être utilisé en mode SSR : les fichiers peuvent exporter une fonction `GET` qui reçoit un [objet de contexte](/fr/reference/api-reference/#endpoint-context) avec des propriétés similaires à celles de l'objet global `Astro`.
```

Comment:
Found an extra whitespace here.
```suggestion
Tout ce qui est décrit dans la section sur les endpoints des fichiers statiques peut également être utilisé en mode SSR: les fichiers peuvent exporter une fonction `GET` qui reçoit un [objet de contexte](/fr/reference/api-reference/#endpoint-context) avec des propriétés similaires à celles de l'objet global `Astro`.
```

Reply:
For me, it's not a question of an extra space. In French, you put a space before a colon. Source:
https://www.projet-voltaire.fr/dossier-voltaire/signes-de-ponctuation-francais/#le-deux-points (cookie warning)

> "Il faut mettre une espace avant et après le deux-points."

In English:
> "Put a space before and after the colon."

Reply:
Oh, right, TIL! Thanks 🙌